.ci/scripts/update-beats.sh

```diff
@@ -5,6 +5,8 @@
 #
 set -euo pipefail
 
+make check-beats-version
```

Comment:
I'm not sure that the branch name is necessarily as expected here either.

How about we introduce another script under `.ci/scripts` which accepts the branch name as an argument? Also, since the check is not generally going to work outside of CI, I would suggest we keep it contained within the script, rather than introducing a Make target. e.g.

```bash
#!/usr/bin/bash
set -xeu
exec grep "BEATS_VERSION?=$BRANCH_VERSION" Makefile
```

The update-beats pipeline (https://github.com/elastic/apm-server/blob/main/.ci/update-beats.groovy) already has a `$BRANCH_VERSION` environment variable we can use. So we would introduce another another `sh` step before the update which calls the script: https://github.com/elastic/apm-server/blob/ad545525e697164a454d4940c30dde12413ec016/.ci/update-beats.groovy#L58

Reply:
I like that! I've pushed a new commit 👍